pkg/scheduler/util/utils.go

```diff
@@ -142,6 +144,12 @@ func IsScalarResourceName(name v1.ResourceName) bool {
 		v1helper.IsPrefixedNativeResource(name) || v1helper.IsAttachableVolumeResourceName(name)
 }
 
+// IsDRAExtendedResourceName returns true when name is an extended resource name, or an implicit extended resource name
+// derived from device class name with the format of deviceclass.resource.kubernetes.io/<device class name>
+func IsDRAExtendedResourceName(name v1.ResourceName) bool {
+	return v1helper.IsExtendedResourceName(name) || strings.HasPrefix(string(name), resourceapi.ResourceDeviceClassPrefix)
```

Comment:
Isn't `deviceclass.resource.kubernetes.io/<device class name>` a string for which `v1helper.IsExtendedResourceName` already returns true?

If yes, then I don't see the point of having `IsDRAExtendedResourceName` because it simply wraps `v1helper.IsExtendedResourceName` and contains dead code.

If no, can you give an example?

Reply:
v1helper.IsExtendedResourceName does not return true for deviceclass.resource.kubernetes.io/<device class name> due to the following code:

// IsPrefixedNativeResource returns true if the resource name is in the
// *kubernetes.io/ namespace.
func IsPrefixedNativeResource(name v1.ResourceName) bool {
	return strings.Contains(string(name), v1.ResourceDefaultNamespacePrefix)
}